packages/extension-api/src/extension-api.d.ts

```diff
@@ -2488,4 +2488,24 @@ declare module '@podman-desktop/api' {
      */
     export function createCliTool(options: CliToolOptions): CliTool;
   }
+
+  export interface ImageCheckPartialResult {
+    category: string;
+    description: string;
+  }
+
+  export interface ImageCheckResult {
+    status: string;
+    partialResults: ImageCheckPartialResult[];
+  }
+
+  export interface ImageCheckerProvider {
+    readonly name: string;
+    readonly categories: string[];
+    checkImage(image: string): Promise<ImageCheckResult>;
+  }
+
+  export namespace imageChecker {
+    export function registerImageCheckerProvider(id: string, imageCheckerProvider: ImageCheckerProvider): Disposable;
```

Comment:
Hello, what is the goal of the id ?

Reply:
To make the provider unique (between extensions, but an extension could also declare several image checker providers). I got inspiration from other providers (`registerAuthenticationProvider`), but it could be created automatically, like `clitool`?

Reply:
got it, I think it's because for Authentication Provider it's one of the few where it needs to provide a public identifier so that other consumers can ask for a 'github' provider

We took inspiration of providers from https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts (you'll see that only few of them provides an identifier, the identifier mostly being the matching part.


Here, as you said, as your goal is for unicity  it needs to be managed internally by the core part.

Reply:
ok, I'll change to compute it internally, thanks